`CodeBlock` docs (HDS-2595) (#1749)
hashicorp/design-system: website/docs/components/code-block/partials/code/component-api.md

## Component API

This component uses [prism.js](https://prismjs.com/) under the hood.

> <Doc::ComponentApi as |C|>

Reviewer: missing `onInput` ?

Author: It was indeed missing, now that we decided to remove the editable feature is not needed anymore.

  </C.Property>
  <C.Property @name="language" @type="string" @values={{array "bash" "go" "hcl" "json" "log" "shell-session" "yaml"}}>
    The coding language to use for syntax highlighting. If you need additional languages <LinkTo class="doc-link-generic" @route="show" @model="about/support">contact the Design Systems Team</LinkTo>.
  </C.Property>
  <C.Property @name="hasCopyButton" @type="boolean" @default="false">
    Used to control whether a copy button for copying the code/text content will be displayed.
  </C.Property>
  <C.Property @name="hasLineNumbers" @type="boolean" @default="true">
    Used to control display of line numbers.
  </C.Property>
  <C.Property @name="hasLineWrapping" @type="boolean" @default="false">
    Used to control line wrapping for lines of code. If `true`, lines of code will wrap to fit the available space. Otherwise, horizontal scrolling is enabled if lines overflow the available space.
  </C.Property>
  <C.Property @name="highlightLines" @type="string">
    Accepts a list or range of line-numbers to highlight. (Examples: `2, 4`, `6-10` )
  </C.Property>

>   <C.Property @name="maxHeight" @type="string" @default="auto">

Reviewer: Probably we should specify that this height doesn't include the header/title/description (it's not obvious)

Author: Addressed in 6fc345cd842f65ee198aed9d55cb47829db24756

    Accepts any valid CSS unit. If the `CodeBlock` content exceeds the maximum height a vertical scrollbar is enabled.
  </C.Property>
  <C.Property @name="contentEditable" @type="boolean" @default="false">

>     Determines whether users can edit the text content of the `CodeBlock`.

Reviewer: We need to explain how we handle the syntax highlighting in relation to being editable

Author: This argument has been removed

  </C.Property>
  <C.Property @name="isStandalone" @type="boolean" @default="true">

>     Alters the visual style of the component by setting rounded borders. When used within another component or when the context requires it, you can turn it off.

Reviewer: ```suggestion
    Applies rounded borders to the component. When used within another component or when the context requires it, you can turn it off.
```
